### PR TITLE
chore(ci): Dispatch documentation notifications upon label added/removed

### PR DESCRIPTION
**Pull request by phated**
# Description

## Problem\*

Closes #1338 (replaces it) 

## Summary\*

This combines the various documentation labeler workflows into a single workflow. It uses fixes from the upstream issue-labeler action and applies https://github.com/github/issue-labeler/pull/71 and https://github.com/github/issue-labeler/pull/72 patches.

This allows us to rely on the outputs of the workflow to know whether we need to open or close an issue on the docs repo.

While working on this I discovered bugs with the docs workflows so this relies on https://github.com/noir-lang/docs/pull/169

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ on:
       - opened
       - reopened
       - edited
-      - synchronize
```
Comment (phated) [inline .github/workflows/labeler.yml:8]: There's no reason to run this on push, so I am removing `synchronize`

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -15,10 +14,31 @@ jobs:
...
-        continue-on-error: true
+      - name: Label pull request
+        uses: phated/issue-labeler@master
+        id: issue-labeler
         with:
-          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
           configuration-path: .github/labeler.yml
           enable-versioned-regex: 0
+          sync-labels: 1
```
Comment (phated) [inline .github/workflows/labeler.yml:24]: The new version of the action has `sync-labels` option which avoids our hacky workaround from before.

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -15,10 +14,32 @@ jobs:
...
+
+      - name: Create documentation issue
+        if: ${{ steps.issue-labeler.outputs.labels-added }}
+        uses: benc-uk/workflow-dispatch@v1
+        with:
+          workflow: new-migrated-issue.yml
+          repo: noir-lang/docs
+          ref: phated/fix-issue-workflows
```
Comment (phated) [inline .github/workflows/labeler.yml:33]:
```suggestion
          ref: master
```
once the PR in docs is merged

File: .github/workflows/labeler.yml [hunk 4]
```diff
@@ -15,10 +14,32 @@ jobs:
...
+          token: ${{ secrets.DOCS_REPO_TOKEN }}
+          inputs: '{ "pr-number": "${{ github.event.pull_request.number }}" }'
+
+      - name: Close documentation issue
+        if: ${{ steps.issue-labeler.outputs.labels-removed }}
+        uses: benc-uk/workflow-dispatch@v1
+        with:
+          workflow: delete-migrated-issue.yml
+          repo: noir-lang/docs
+          ref: phated/fix-issue-workflows
```
Comment (phated) [inline .github/workflows/labeler.yml:43]:
```suggestion
          ref: master
```
once the PR in docs is merged

**Review by Savio-Sou — DISMISSED** [review]
Thanks for putting it tgt. The labeling part LGTM.

@signorecello feel free to merge the doc repo PR and this PR if they look ok to you.

**Review by signorecello — APPROVED** [review]

**Pull request merged**